development-guide/source-connector-development-guide.md

```diff
@@ -21,6 +21,28 @@ The `Schema` RPC call retrieves the user's schemas, tables, and columns. It also
 ### Update
 The `Update` RPC call should retrieve data from the source. We send a request using the `UpdateRequest` message, which includes the user's connection state, credentials, and schema information. The response, streaming through the `UpdateResponse` message, can contain data records and other supported operations.
 
+### Record types
+
+#### Upsert
+The `upsert` record type essentially translates to a delete + insert SQL operation, ie, if a row with that primary key is already present in the destination, it will first be deleted then re-inserted. If the row with that primary key does not exist in the destination, it boils down to a simple insert.
```

Comment:
@rajat-kushwaha upsert is an operation type that is applied to a record, not a record type per se, isn't it?

Reply:
I see that we might classify the records by the applied operation.
Can we have this clarification in the heading, like "Record types as defined by operation"?

Reply:
Also, all instances of "ie" should be "i.e." = "id est"